keps/6757-failure-recovery/README.md

```diff
@@ -0,0 +1,379 @@
+# KEP-6757: Failure Recovery
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [Replacing Pods Which Are Still Running](#replacing-pods-which-are-still-running)
+- [Design Details](#design-details)
+  - [Affected Pods](#affected-pods)
+  - [Grace Period](#grace-period)
+  - [Implementation Overview](#implementation-overview)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration Tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+  - [Beta](#beta)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Do Nothing](#do-nothing)
+  - [Control All Pods Managed By Kueue](#control-all-pods-managed-by-kueue)
+  - [Introduce a <code>FailureRecoveryPolicy</code> API](#introduce-a-failurerecoverypolicy-api)
+  - [Managing The <code>node.kubernetes.io/out-of-service</code> Taint On <code>Node</code>](#managing-the-nodekubernetesioout-of-service-taint-on-node)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces an opt-in mechanism of timeout-based graceful handling of zombie pods,
+i.e. pods that are stuck in the `Pending`/`Running` state due to a node-level failure.
+
+## Motivation
+
+Currently, a malfunction of the `kubelet` (or a more general node failure) results in the pods running on that node to become stuck.
+When the `kubelet` fails to send its regular heartbeat to the control plane within `node-monitor-grace-period`, the node is deemed unhealthy and is assigned the `node.kubernetes.io/unreachable` taint. By default, Kubernetes [automatically adds a 5-minute toleration](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions) for pods running in the affected node (this can be explicitly set by a different value by the user/controller).
```

Comment:
```suggestion
When the `kubelet` fails to send its regular heartbeat to the control plane within `node-monitor-grace-period`, 
the node is deemed unhealthy and is assigned the `node.kubernetes.io/unreachable` taint (after being marked with `node.kubernetes.io/not-ready` first).
By default, Kubernetes [automatically adds a 5-minute toleration](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions) for pods running in the affected node (this can be explicitly set by a different value by the user/controller).
```

AFAIK, the node taints will be progressing N/A -> `node.kubernetes.io/not-ready` -> `node.kubernetes.io/unreachable`

Reply:
For reference, it's described [here](https://kubernetes.io/docs/reference/node/node-status/#condition) in the docs.